Remove TRUNCATE PARTITION as non-supported (#13868) (#13880)
pingcap/docs-cn: mysql-compatibility.md

* TiDB 中，`ALGORITHM={INSTANT,INPLACE,COPY}` 语法只作为一种指定，并不更改 `ALTER` 算法，详情参阅 [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md)。
* 不支持添加或删除 `CLUSTERED` 类型的主键。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
* 不支持指定不同类型的索引 (`HASH|BTREE|RTREE|FULLTEXT`)。若指定了不同类型的索引，TiDB 会解析并忽略这些索引。

> * 分区表支持 `HASH`、`RANGE`、`LIST` 分区类型。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type, treat as normal table`。

Reviewer: ```suggestion
* 分区表支持 `HASH`、`RANGE`、`LIST` 分区类型。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type, treat as normal table`。
```
```suggestion
* 分区表支持 `HASH`、`RANGE`、`LIST` 分区类型。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type, treat as normal table`。
* 分区表支持 `ADD`、`DROP`、`TRUNCATE` 操作，其他分区操作会被忽略。TiDB 不支持以下分区表语法：
```

* TiDB 中，`ALGORITHM={INSTANT,INPLACE,COPY}` 语法只作为一种指定，并不更改 `ALTER` 算法，详情参阅 [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md)。
* 不支持添加或删除 `CLUSTERED` 类型的主键。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
* 不支持指定不同类型的索引 (`HASH|BTREE|RTREE|FULLTEXT`)。若指定了不同类型的索引，TiDB 会解析并忽略这些索引。
* 分区表支持 `HASH`、`RANGE`、`LIST` 分区类型。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type, treat as normal table`。

>     + `PARTITION BY LIST`

Reviewer: ```suggestion
```

> * 分区表支持 Hash、Range 和 `Add`/`Drop`/`Truncate`/`Coalesce`。其他分区操作将被忽略，可能会返回 `Warning: Unsupported partition type, treat as normal table`。不支持以下分区表语法：

Reviewer: ```suggestion
* 分区表支持 Hash、Range 和 `Add`/`Drop`/`Truncate`/`Coalesce`。其他分区操作将被忽略，会返回 `Warning: Unsupported partition type, treat as normal table`。不支持以下分区表语法：
```

> * 分区表支持 Hash、Range 和 `Add`/`Drop`/`Truncate`/`Coalesce`。其他分区操作将被忽略，可能会返回 `Warning: Unsupported partition type, treat as normal table`。不支持以下分区表语法：

Reviewer: ```suggestion
* 分区表支持 `HASH`、`RANGE`、`LIST` 分区类型。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type, treat as normal table`。
```

    + `PARTITION BY KEY`
    + `SUBPARTITION`

>     + `{CHECK|EXCHANGE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE} PARTITION`

Reviewer: ```suggestion
    + `{CHECK|EXCHANGE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION`
```

>     + `{CHECK|EXCHANGE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION`

Reviewer: ```suggestion
    + `{CHECK|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION`
```

* TiDB 中，`ALGORITHM={INSTANT,INPLACE,COPY}` 语法只作为一种指定，并不更改 `ALTER` 算法，详情参阅 [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md)。
* 不支持添加或删除 `CLUSTERED` 类型的主键。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
* 不支持指定不同类型的索引 (`HASH|BTREE|RTREE|FULLTEXT`)。若指定了不同类型的索引，TiDB 会解析并忽略这些索引。

> <<<<<<< HEAD
> * 分区表支持 Hash、Range 和 `Add`/`Drop`/`Truncate`/`Coalesce`。其他分区操作将被忽略，可能会报 `Warning: Unsupported partition type, treat as normal table` 错误。不支持以下分区表语法：
>     + `PARTITION BY LIST`
>     + `PARTITION BY KEY`
>     + `SUBPARTITION`
>     + `{CHECK|EXCHANGE|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE} PARTITION`
> =======
> * 分区表支持 `HASH`、`RANGE`、`LIST` 和 `KEY` 分区类型。`KEY` 分区类型暂不支持分区字段列表为空的语句。对于不支持的分区类型，TiDB 会返回 `Warning: Unsupported partition type %s, treat as normal table`，其中 `%s` 为不支持的具体分区类型。
> * 分区表还支持 `ADD`、`DROP`、`TRUNCATE`、`REORGANIZE` 操作，其他分区操作会被忽略。TiDB 不支持以下分区表语法：
>     + `SUBPARTITION`
>     + `{CHECK|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|COALESCE} PARTITION`
> 
>   更多详情，请参考[分区表文档](/partitioned-table.md)。
> >>>>>>> 335efcc17c (Remove TRUNCATE PARTITION as non-supported (#13868))

Reviewer: ```suggestion
* 分区表支持 Hash、Range 和 `Add`/`Drop`/`Truncate`/`Coalesce`。其他分区操作将被忽略，可能会返回 `Warning: Unsupported partition type, treat as normal table`。不支持以下分区表语法：
    + `PARTITION BY LIST`
    + `PARTITION BY KEY`
    + `SUBPARTITION`
    + `{CHECK|EXCHANGE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE} PARTITION`
```
